test/tests/functional/pbs_cpuset_reliable_job_startup.py

```diff
@@ -0,0 +1,162 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2018 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# The PBS Pro software is licensed under the terms of the GNU Affero General
+# Public License agreement ("AGPL"), except where a separate commercial license
+# agreement for PBS Pro version 14 or later has been executed in writing with
+# Altair.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under a
+# commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+import time
+from tests.functional import *
+
+
+class TestPbsReliableJobStartupOnCpuset(TestFunctional):
+
+    """
+    This tests the Reliable Job Startup Feature with cpuset mom.
+    A job can be started with extra nodes with node failures tolerated
+    during job start. Released cpuset resources can be reused by another job.
+    """
+
+    def setUp(self):
+        if not self.mom.is_cpuset_mom():
+            self.skipTest("Test suite only meant to run with cpuset mom.")
+        TestFunctional.setUp(self)
+
+        self.vnA = self.server.shortname + '[0]'
+        rv = self.server.status(NODE, 'resources_available.ncpus',
+                                id=self.vnA)
+        self.vnA_ncpus = int(rv[0]['resources_available.ncpus'])
```

Comment:
I have added a code comment. Since test machines would have varying amount of cpus per vnode I wanted to find out the number of cpus so that the test can run on various test machines.

Reply:
that doesn't answer my question :) "why do we need to submit jobs asking all ncpus on a vnode?"

Reply:
The test needs to initially request for 2 chunks that would span 2 vnodes.  I have changed the test so it requests 1 ncpu per vnode, place=vscatter.